src/wordpress-playground/index.tsx

```diff
@@ -297,3 +299,104 @@ function zipNameToHumanName(zipName) {
 }
 
 main();
+
+async function overwriteFile() {
+	const targetFile = document.getElementById('target-file').value;
+	const targetContent = document.getElementById('target-content').value;
+	console.log(targetFile);
+	console.log(targetContent);
+	await workerThread.writeFile(targetFile, targetContent);
+}
+
+async function generateZip() {
+	await workerThread.run({
+		code: `
+			<?php
+				$zip = new ZipArchive;
+				$res = $zip->open('/wordpress-playground-export.zip', ZipArchive::CREATE);
+				if ($res === TRUE) {
+					$directories = array();
+					$directories[] = '/wordpress/';
+
+					while(sizeof($directories)) {
+						$dir = array_pop($directories);
+
+						if ($handle = opendir($dir)) {
+
+							while (false !== ($entry = readdir($handle))) {
+								
+								if ($entry == '.' || $entry == '..') {
+									continue;
+								}
+
+								$entry = $dir . $entry;
+
+								if (is_dir($entry)) {
+
+									$directory_path = $entry . '/';
+									array_push($directories, $directory_path);
+
+								} elseif (is_file($entry)) {
+
+									$zip->addFile($entry);
+								}
+							}
+							closedir($handle);
+						}
+					}
+					$zip->close();
+				}
+			?>
+		`,
+	});
+	const fileBuffer = await workerThread.readFileAsBuffer(
```

Comment:
Awesome! 🙌  Is this reasonably fast, as in no significant delay after pressing the export button? My local repo is a mess at the moment and I'd rather not stash everything atm.

Reply:
It works pretty quickly for me. I'd say we're good to go with this approach.